packages/proto-rpc/src/RpcCommunicator.ts

```diff
@@ -0,0 +1,278 @@
+import * as Err from './errors'
+import { ErrorCode } from './errors'
+import {
+    ClientTransport,
+    ResultParts,
+    ProtoRpcOptions,
+    Event as DhtTransportClientEvent
+} from './ClientTransport'
+import {
+    RpcMessage,
+    RpcResponseError
+} from './proto/ProtoRpc'
+import { Empty } from './proto/google/protobuf/empty'
+import { CallContext, ServerRegistryEvent as ServerRegistryEvent, Parser, Serializer, ServerRegistry } from './ServerRegistry'
+import { EventEmitter } from 'events'
+import { DeferredState } from '@protobuf-ts/runtime-rpc'
+import { ServerCallContext } from '@protobuf-ts/runtime-rpc'
+import { Logger } from './Logger'
+
+export enum StatusCodes {
+    OK = 'OK',
+    STOPPED = 'STOPPED',
+    DEADLINE_EXCEEDED = 'DEADLINE_EXCEEDED',
+    SERVER_ERROR = 'SERVER_ERROR'
+}
+
+export enum RpcCommunicatorEvents {
+    OUTGOING_MESSAGE = 'streamr:proto-rpc:rpc-communicator:outgoing-message',
+}
+
+export interface RpcCommunicatorConfig {
+    rpcRequestTimeout?: number,
+}
+
+interface IRpcIo {
+    handleIncomingMessage(message: Uint8Array, callContext?: CallContext): Promise<void> 
+    on(event: RpcCommunicatorEvents.OUTGOING_MESSAGE, listener: (message: Uint8Array, callContext?: CallContext) => void): this
+}
+
+interface OngoingRequest {
+    deferredPromises: ResultParts,
+    timeoutRef: NodeJS.Timeout
+}
+
+const logger = new Logger(module)
+
+export class RpcCommunicator extends EventEmitter implements IRpcIo {
+    private stopped = false
+    private readonly rpcClientTransport: ClientTransport
+    private readonly rpcServerRegistry: ServerRegistry
+    private readonly ongoingRequests: Map<string, OngoingRequest>
+    private rpcRequestTimeout = 5000
+
+    constructor(params?: RpcCommunicatorConfig) {
+        super()
+        if (params && params.hasOwnProperty('rpcRequestTimeout')) {
+            this.rpcRequestTimeout = params.rpcRequestTimeout!
+        }
```

Comment:
```suggestion
        this.rpcRequestTimeout = params?.rpcRequestTimeout ?? 5000
```

And we can remove the default value from line 52